lib/matplotlib/axis.py

```diff
@@ -1523,6 +1523,23 @@ def set_ticklabels(self, ticklabels, *args, **kwargs):
 
         ACCEPTS: sequence of strings
         """
+
+        #######################################################################
+        # check whether the input is return value of get_ticklabels()
+        # get_ticklabels() return a cbook.silent_list of Text objects  
+        if isinstance(ticklabels, cbook.silent_list):
```

Comment:
Using `isintance` is usually an anti-pattern.  The input is supposed to be a sequence of strings, so we can assume we can iterate over it.  I think something like

```python
new_labels = []
for t in tick_labels:
    try:
        new_labels.append(t.get_text())
    except AttributeError:
       new_labels.append(t)

ticklabels = new_labels
```

Would be more robust and pythonic.